src/OrbitGl/TimeGraph.cpp

```diff
@@ -164,6 +164,14 @@ void TimeGraph::ZoomTime(float zoom_value, double mouse_ratio) {
   double min_time_us = ref_time_us_ - time_left / scale;
   double max_time_us = ref_time_us_ + time_right / scale;
 
+  const double duration = max_time_us - min_time_us;
+
+  if (duration < kTimeGraphMinTimeWindowsUs) {
+    const double diff = kTimeGraphMinTimeWindowsUs - duration;
+    min_time_us -= diff * mouse_ratio;
+    max_time_us += diff * (1. - mouse_ratio);
```

Comment:
1.f

Reply:
Can you explain why? 1.f will be promoted to `double` anyway, right?

Reply:
sorry, you are right. We are using doubles here.